tests/accumulator/tests.c

```diff
@@ -0,0 +1,210 @@
+#include "utils/mempool.h"
+#include "utils/bitmask.h"
+#include "collect.h"
+#include "packetsgen.h"
+
+#define NB_PKTS 60
+
+static void realloc_pkts(struct rte_mbuf **packets)
+{
+	for (int i = 0; i < NB_PKTS; i++) {
+		packets[i] = rte_pktmbuf_alloc(mp);
+		g_assert(packets[i]);
+		packets[i]->udata64 = i;
+	}
+}
+
+static void accumulator_packets_poll(enum pg_side dir, uint64_t loop)
+{
+	struct pg_brick *acc, *col1, *col2;
+	struct pg_error *error = NULL;
+	struct rte_mbuf *packets[NB_PKTS];
+	struct rte_mbuf **res_pkts[PG_MAX_SIDE];
+	uint16_t packet_count;
+	uint64_t filtered_mask;
+
+	memset(packets, 0, NB_PKTS * sizeof(struct rte_mbuf *));
+	realloc_pkts(packets);
+	g_assert(!error);
+	acc = pg_accumulator_new("acc", dir, loop, &error);
+	g_assert(!error);
+	col1 = pg_collect_new("col1", &error);
+	g_assert(!error);
+	col2 = pg_collect_new("col2", &error);
+	g_assert(!error);
+		pg_brick_chained_links(&error, col1, acc, col2);
+		g_assert(!error);
+	if (dir == PG_WEST_SIDE || dir == PG_MAX_SIDE)
+		pg_brick_burst_to_east(acc, 0, packets,
+				       pg_mask_firsts(NB_PKTS), &error);
+	if (dir == PG_EAST_SIDE || dir == PG_MAX_SIDE)
+		pg_brick_burst_to_west(acc, 0, packets,
+				       pg_mask_firsts(NB_PKTS), &error);
+	for (uint64_t i = 0; i < loop + 1; i++) {
+		pg_brick_poll(acc, &packet_count, &error);
+		g_assert(!error);
+	}
+	res_pkts[0] = pg_brick_west_burst_get(col1, &filtered_mask, &error);
+	g_assert(!error);
+	res_pkts[1] = pg_brick_east_burst_get(col2, &filtered_mask, &error);
+	g_assert(!error);
+
+	if (dir == PG_WEST_SIDE || dir == PG_MAX_SIDE) {
+		g_assert(res_pkts[0]);
+		if (dir == PG_WEST_SIDE)
+			g_assert(!res_pkts[1]);
+		for (uint64_t i = 0; i < NB_PKTS; i++)
+			g_assert(res_pkts[0][i]->udata64 == i % NB_PKTS);
+	}
+	if (dir == PG_EAST_SIDE || dir == PG_MAX_SIDE) {
+		g_assert(res_pkts[1]);
+		if (dir == PG_EAST_SIDE)
+			g_assert(!res_pkts[0]);
+		for (uint64_t i = 0; i < NB_PKTS; i++)
+			g_assert(res_pkts[1][i]->udata64 == i % NB_PKTS);
+	}
+	pg_brick_destroy(col1);
+	pg_brick_destroy(col2);
+	pg_brick_destroy(acc);
+}
+
+static void test_accumulator_poll(void)
+{
+	accumulator_packets_poll(PG_WEST_SIDE, 5);
+	accumulator_packets_poll(PG_EAST_SIDE, 0);
+	accumulator_packets_poll(PG_MAX_SIDE, 23462);
+}
+
+uint64_t pg_res_id;
+
+static void lanch_burst(struct pg_brick *acc,
+			struct pg_brick *col1,
+			struct pg_brick *col2,
+			struct rte_mbuf **packets,
+			uint64_t expected,
+			enum pg_side dir)
+{
+	struct rte_mbuf **result_pkts;
+	uint64_t pkts_mask = 0;
+	struct pg_error *error = NULL;
+	uint64_t i;
+
+	if (dir == PG_WEST_SIDE || dir == PG_MAX_SIDE) {
+		pg_brick_burst_to_west(acc, 0, packets,
+				       pg_mask_firsts(NB_PKTS), &error);
+		g_assert(!error);
+		result_pkts = pg_brick_east_burst_get(col1, &pkts_mask, &error);
+		for (i = 0; i < expected; i++)
+			g_assert(result_pkts[i]->udata64 ==
+				 (i + pg_res_id) % NB_PKTS);
+	} else if (dir == PG_EAST_SIDE || dir == PG_MAX_SIDE) {
```

Comment:
does it work with PG_MAX_SIDE ?
I mean using `else if` instead of if seems very suspicious 
also, refactor but chunk of code, into one function ?